rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -136,6 +175,13 @@ def _entity_data(event: UserUttered) -> Dict[Text, Any]:
 
     @staticmethod
     def _remove_user_text_if_intent(trackers_as_states: List[List[State]]) -> None:
+        """Deletes user text from state dictionaries if intent is present.
```

Comment:
I think I'm missing some explanation either here or wherever this method is called of what the idea behind removing user text is. Could we add something explanatory, probably in this docstring?

Reply:
@dakshvar22 perhaps you can explain the why behind this one? I think it has to do with not featurizing the utterance text if we are using intents as the representation of a user utterance, but not totally sure.

Reply:
> I think it has to do with not featurizing the utterance text if we are using intents as the representation of a user utterance

Yes 👍 During training if intent was present, then we discard text. During inference, if prediction was made based on intent during any conversation step, we remove text.

Reply:
@kedz Can you take care of this in your PR 9005 itself?

Reply:
Addressed in #9005. Will resolve this conversation when that PR has been merged into `intent-ted`.